### ISSUE TITLE
Allow attribute does not surpress unknown_or_malformed_diagnostic_attributes

### ISSUE THREAD

**Issue by SpecificProtagonist** [rust-lang/rust-clippy#14044]
### Summary

Discovered in https://github.com/bevyengine/bevy/pull/17441

### Lint Name

unknown_or_malformed_diagnostic_attributes

### Reproducer

I tried [this code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6c8f86f28de22fc5299063c8a9720e61):

```rust
trait _Trait {}
#[allow(unknown_or_malformed_diagnostic_attributes)]
#[diagnostic::abcdef]
impl _Trait for () {}
```

I saw this happen:

```
warning: unknown diagnostic attribute
 --> src/lib.rs:3:15
  |
3 | #[diagnostic::abcdef]
  |               ^^^^^^
  |
  = note: `#[warn(unknown_or_malformed_diagnostic_attributes)]` on by default

warning: `playground` (lib) generated 1 warning
```

I expected to see this happen:

No warnings.

### Version

```text
rustc 1.86.0-nightly (9a1d156f3 2025-01-19)
binary: rustc
commit-hash: 9a1d156f38c51441ee51e5a068f1d0caf4bb0f27
commit-date: 2025-01-19
host: x86_64-unknown-linux-gnu
release: 1.86.0-nightly
LLVM version: 19.1.7
```

**Comment by samueltardieu** [comment]
@SpecificProtagonist This is a rustc lint, not a Clippy lint (it is not prefixed with `clippy::`). You will get the warning just by compiling your file.

It looks like you need to either allow the lint globally (`#![…]`), or apply it to a scope such as a module. At last in those cases it will not issue the warning.
